### ISSUE TITLE
Add BucketLookupType for s3 bucket

### ISSUE THREAD

**Issue by devmentality** [grafana/pyroscope#3451]
#### Is your feature request related to a problem? Please describe.

Hi, Pyroscope Team!
I encountered a problem with s3 configuration. We have an on-premise Ceph installation as our company's s3 provider. It supports only virtual-host-style requests. But it seems that underlying minio library, which is used in Pyroscope, chooses to use path-style requests. 

Why minio-go uses path-style requests in our case? 
By default Pyroscope uses AutoLookup strategy, which resolves to VirtualHostLookup or PathLookup by minio-go library. As I can see from the [function](https://github.com/minio/minio-go/blob/master/pkg/s3utils/utils.go#L71), the library chooses virtual-host style only for AWS, Google and Alibaba provider. It means that the library's AutoLookup strategy will choose path-style requests for my on-premise s3.

#### Describe the solution you'd like

It would be great to have `s3.bucket_lookup_type` configuration parameter to force Pyroscope use virtual-host-style requests. 
Example of such a parameter in Mimir: https://github.com/grafana/mimir/pull/7684/files#diff-8a8879ab8601d8d13855963d239c72b9f082bd2be2889bbbce3282d05fcddf7d. 

#### Additional context

I see that Pyroscope already has `s3.force-path-style` flag, which is has the opposite meaning to what we need. 
I can help with this feature by creating a PR, but we need to discuss how new parameter will get along with `s3.force-path-style` first.

**Comment by simonswine** [comment]
@devmentality yes I agree with your suggestions and would more than welcome a PR from you ❤️ 

Not fully sure if this has drawbacks that I missed but, I would make sure:

* Error when force-path is true and lookup-type is `virtual-host`.
* To display a warning if force-path is set to true, that this flag is deprecated and might be removed going forward. We then use lookup mode `path` internally.

**Comment by lzyan** [comment]
We also need this feature，hope to be supported by offical

**Comment by dozer47528** [comment]
Same problem

**Comment by devmentality** [comment]
Hi @simonswine!
Implemented this feature in the mentioned above PR

**Comment by simonswine** [comment]
Yes thanks for your PR, is released as v1.12.0
